scrapers_next/pa/committees.py

```diff
@@ -1,105 +1,603 @@
+            elif subcom_name == "Cities, Third Class":
+                build_committee("Cities", "Chair, Third Class")
+            else:
+                build_committee(subcom_name, "Chair")
+
+        yield from subcoms.values()
+
+
+# Helper function for CommitteeDetail, gets name and role from a member element
+def extract_member_data(base_element):
+
+    name = XPath(".//a/text()").match_one(base_element)
+    role = None
+    try:
+        role = XPath("./div[@class='position']/text()").match_one(base_element)
+    except SelectorError:
+        role = "Member"
+
+    # Convert from "lastname, firstname" to "firstname lastname"
+    name = name.strip().split(", ")
+    name.reverse()
+    name = " ".join(name)
+
+    # Roles need to be stripped and have a special character substituted
+    role = role.strip().replace("\u2011", "-")
+
+    return name, role
+
+
+class CapitolPreservationComm(HtmlPage):
+    source = "http://cpc.state.pa.us/about/capitol-preservation-committee-members.cfm"
+
+    def process_page(self):
+        com = ScrapeCommittee(
+            name="Capitol Preservation",
+            classification="committee",
+            chamber="legislature",
+        )
+        com.add_link("http://cpc.state.pa.us/", note="Homepage")
+        com.add_source(
+            self.source.url,
+            note="Member list page",
+        )
+
+        members = XPath(
+            "//div[@class='journal-content-article']/div[@class='member']"
         ).match(self.root)
-        for mem in minority_members:
+
+        for member in members:
+            # Skip vacant seats
+            image = XPath("./img").match_one(member)
+            if image.get("src").endswith("/images/about/members/Vacant.jpg"):
+                continue
+
+            # These titles are not all formatted in the same way
+            title = XPath("./p/text()").match(member)
+
+            # First line is always the title (since vacant positions are skipped already)
+            name = title[0]
+
+            # If the final character of the name is a comma, remove it
+            if name.endswith(","):
+                name = name[:-1]
+
+            title = " ".join(title[1:])
+            # Split title by comma, remove blank elements
+            title = [x.strip() for x in title.split(",") if x.strip()]
+
+            # Search the text for these titles and
+            # set their role to the corresponding value
+            role = "Member"
+            roles = {
+                "Representative": "Member",
+                "Senator": "Member",
+                "Emeritus Member": "Emeritus Member",
+                "Governor's Appointee": "Governor's Appointee",
+                "Supreme Court Appointee": "Supreme Court Appointee",
+                "Executive Director": "Executive Director",
+            }
+            for k, v in roles.items():
+                if k in title:
+                    role = v
```

Comment:
This was very thorough, and thank you for that, but we want a bit more role information than what it is currently getting. I have a proposed solution.

**Issue:**
- Currently, [from the page being scraped](http://cpc.state.pa.us/about/capitol-preservation-committee-members.cfm), Andrea Bakewell is only being stored with role as "Executive Director", but we don't know of what. 
- Similar for Frank E. Dittenhafer, II: we are getting "Governor's Appointee" but not "FAIA, LEED AP".

**Solution for handling members:**
- if there is the placeholder image (indicating vacancy), then skip adding that member
- if the `p` for that members includes a `strong`, then get the text in the `strong` as the role, the text after the `strong` but before the comma or next `br` as the name, and then any any following text gets concatenated to the role with a comma and space (so John R. Bowie would have role as "Vice Chairman, Governor's Appointee") unless the substring that would be concatenated would be either "Senator" or "Representative" (as in the case of Treasurer Patty Kim)
- for the remainder of the members, all text after the comma / `br` should be added as role (to fix issues like in the cases of Blackwell and Dittenhafer) unless it is "Senator" or "Representative" (you already did a great job of handling Dittenhafer's case when it came to the comma in his name, so just make sure this still handles the ", II" as the end of his name and not part of the role